hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/store/CachedRecordStore.java

```diff
@@ -125,7 +125,6 @@ public boolean loadCache(boolean force) throws IOException {
       } catch (IOException e) {
         LOG.error("Cannot get \"{}\" records from the State Store",
             getRecordClass().getSimpleName());
-        this.initialized = false;
```

Comment:
Actually, this is the most important line in the patch. :)

If initialized goes to false, it will block serving up the contents of the cache. With the patch the router will be up until dfs.federation.router.store.membership.expiration expires, in which case the router goes into safe mode.